examples/specs_integration/src/sync/entity_create.rs

```diff
@@ -0,0 +1,122 @@
+                templates,
+                spatials,
+                default_parent: _,
+            } = self;
+
+            unsafe {
+                let parent = if let Some(parent) = grandparent.find_node(
+                    GodotString::from_str(gd_sceneloc.parent_name),
+                    false,
+                    false,
+                ) {
+                    parent
+                } else {
+                    // TODO: Create parent here.
+
+                    let mut new_parent = Spatial::new()
+                        .cast()
+                        .expect("Spatial is a hardcoded subclass of Node! WTF!?");
+                    grandparent.add_child(Some(new_parent), false);
+                    new_parent.set_name(GodotString::from_str(gd_sceneloc.parent_name));
+                    new_parent
+                };
+
+                // TODO: Allow moving the spatial
+                let gd_spatial = gd_spatial_entry.or_insert_with(|| {
+                    // If a this specs entity has no spatial yet...
+                    let mut new_spatial = GdEntityCreation::instance_template(
+                        parent,
+                        gd_sceneloc.template,
+                        templates,
+                    );
+
+                    new_spatial.set_name(GodotString::from_str(gd_sceneloc.child_name.clone()));
+                    let old_spatial = spatials.insert(entity.id(), new_spatial);
+                    old_spatial.map(|mut spatial| spatial.queue_free());
+
+                    GdSpatial {
+                        curr_template: gd_sceneloc.template,
+                    }
+                });
+
+                // If a previous frame already created a template.
+                if gd_spatial.curr_template != gd_sceneloc.template {
+                    let mut new_spatial = GdEntityCreation::instance_template(
+                        parent,
+                        gd_sceneloc.template,
+                        templates,
+                    );
+
+                    let old_spatial = spatials.insert(entity.id(), new_spatial);
+                    old_spatial.map(|mut spatial| spatial.queue_free());
+
+                    gd_spatial.curr_template = gd_sceneloc.template;
+
+                    new_spatial.set_name(GodotString::from_str(gd_sceneloc.child_name.clone()));
+                    // Add code which moves from old world to new world...
+                }
+            }
+        }
+    }
+}
+
+impl GdEntityCreation<'_> {
+    unsafe fn instance_template(
+        mut parent: Node,
+        template_name: Option<Template3dName>,
+        templates: &HashMap<Template3dName, Template3dEntity>,
+    ) -> Spatial {
+        let template = template_name.and_then(|template_name| templates.get(&template_name));
+
+        let new_spatial = match template {
+            Some(Template3dEntity::Scene(packed_scene)) => {
+                let scene_opt = instance_scene::<Spatial>(packed_scene).ok();

```

Comment:
`instance_scene` is only used here (in the example). Maybe move it to this module?